src/pipeline/src/manager.rs

```diff
@@ -40,8 +43,88 @@ pub mod util;
 /// When it used in cache key, it will be converted to i64 meaning the number of nanoseconds since the epoch.
 pub type PipelineVersion = Option<TimestampNanosecond>;
 
-/// Pipeline info. A tuple of timestamp and pipeline reference.
-pub type PipelineInfo = (Timestamp, PipelineRef);
+/// Unique reference to pipeline name.
+/// A valid pipeline name should be in the format of [<schema>.]<name>
+/// The optional schema is for cross-schema reference.
+/// Note the version can be None while query, which means the latest version of the pipeline.
+#[derive(Debug, Clone)]
+pub struct PipelineName {
+    pub name: String,
+    pub schema: String,
+    pub version: PipelineVersion,
+}
+
+impl PipelineName {
+    /// use [`from_name_and_version`] outside of this crate
+    pub(crate) fn new(name: String, schema: String, version: PipelineVersion) -> Self {
+        Self {
+            name,
+            schema,
+            version,
+        }
+    }
+
+    pub fn from_name_and_version(
+        name: &str,
+        version_str: Option<&str>,
+        default_schema: String,
+    ) -> Result<Self> {
+        let version = to_pipeline_version(version_str)?;
+
+        ensure!(
+            !name.is_empty(),
+            InvalidPipelineNameSnafu {
+                reason: "pipeline name cannot be empty",
+            }
+        );
+
+        let parts = name.split('.').collect::<Vec<&str>>();
+        // if schema is not provided, use the schema from ctx
+        match parts.len() {
+            1 => Ok(Self {
+                name: name.to_string(),
+                schema: default_schema,
+                version,
+            }),
+            2 => Ok(Self {
+                name: parts[1].to_string(),
+                schema: parts[0].to_string(),
+                version,
+            }),
+            _ => InvalidPipelineNameSnafu {
+                reason: "pipeline name must be in the format of [<schema>.]<name>",
+            }
+            .fail()?,
+        }
+    }
+
+    pub fn check_internal_name(&self) -> Result<()> {
+        ensure!(
+            !self
+                .name
+                .starts_with(GREPTIME_INTERNAL_PIPELINE_NAME_PREFIX),
+            InvalidPipelineNameSnafu {
+                reason: "custom pipeline name cannot start with 'greptime_'",
+            }
+        );
+        Ok(())
+    }
+
+    pub fn check_schema(&self, ctx_schema: &str) -> Result<()> {
```

Comment:
Adds a document for public functions.
Notice that it is only used when adding pipelines.